tensorflow/python/ops/array_ops.py

```diff
@@ -3926,6 +3926,77 @@ def reverse_sequence(input,
                      name=None,
                      seq_dim=None,
                      batch_dim=None):
+  """Reverses variable length slices.
+
+  This op first slices `input` along the dimension `batch_axis`, and for
+  each slice `i`, reverses the first `seq_lengths[i]` elements along the
+  dimension `seq_axis`.
+
+  The elements of `seq_lengths` must obey `seq_lengths[i] <=
+  input.dims[seq_dim]`, and `seq_lengths` must be a vector of length
+  `input.dims[batch_dim]`.
+
+  The output slice `i` along dimension `batch_axis` is then given by
+  input slice `i`, with the first `seq_lengths[i]` slices along
+  dimension `seq_axis` reversed.
+
+  For example:
+
+  ```python
+  # Given this:
+  batch_dim = 0
+  seq_dim = 1
+  input.dims = (4, 8, ...)
+  seq_lengths = [7, 2, 3, 5]
+
+  # then slices of input are reversed on seq_dim, but only up to seq_lengths:
+  output[0, 0:7, :, ...] = input[0, 7:0:-1, :, ...]
+  output[1, 0:2, :, ...] = input[1, 2:0:-1, :, ...]
+  output[2, 0:3, :, ...] = input[2, 3:0:-1, :, ...]
+  output[3, 0:5, :, ...] = input[3, 5:0:-1, :, ...]
+
+  # while entries past seq_lens are copied through:
+  output[0, 7:, :, ...] = input[0, 7:, :, ...]
+  output[1, 2:, :, ...] = input[1, 2:, :, ...]
+  output[2, 3:, :, ...] = input[2, 3:, :, ...]
+  output[3, 2:, :, ...] = input[3, 2:, :, ...]
+  ```
+
+  In contrast, if:
+
+  ```python
```

Comment:
As Alex said, it would be nice to have this in the [doctest](https://docs.python.org/3/library/doctest.html) format.

Please remove the ```python and ``` after you add `>>>`.

Try the code out locally on your machine with TF2 rc2 and python3.